client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx

```diff
@@ -81,213 +98,198 @@ beforeEach(() => {
-      <Router>
-        <Review locked={false} prevStage={prevStage} refresh={jest.fn()} />
-      </Router>
-    )
-    await waitFor(() => expect(apiMock).toHaveBeenCalled())
-    expect(container).toMatchSnapshot()
+    const expectedCalls = [
+      apiCalls.getEmptyJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getTargetedContests,
+      apiCalls.getSampleSizeOptions,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      const { container } = renderView()
+      await screen.findByText('Review & Launch')
+      expect(container).toMatchSnapshot()
+    })
   })
 
   it('renders despite missing jurisdictions on opportunistic contest', async () => {
-    apiMock.mockImplementation(
-      generateApiMock(
-        sampleSizeMock,
-        contestMocks.filledOpportunisticWithJurisdictionId,
-        {
-          jurisdictions: [],
-        }
-      )
-    )
     auditSettingsMock.mockReturnValue(settingsMock.full)
-    const { container } = render(
-      <Router>
-        <Review locked={false} prevStage={prevStage} refresh={jest.fn()} />
-      </Router>
-    )
-    await waitFor(() => expect(apiMock).toHaveBeenCalled())
-    expect(container).toMatchSnapshot()
+    const expectedCalls = [
+      apiCalls.getEmptyJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getOpportunisticContests,
+      apiCalls.getSampleSizeOptions,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      const { container } = renderView()
+      await screen.findByText('Review & Launch')
+      expect(container).toMatchSnapshot()
+    })
   })
 
   it('launches the first round', async () => {
-    apiMock.mockImplementation(
-      generateApiMock(
-        sampleSizeMock,
-        contestMocks.filledTargetedWithJurisdictionId,
-        { jurisdictions: jurisdictionMocks.allManifests }
-      )
-    )
-    render(
-      <Router>
-        <Review locked={false} prevStage={prevStage} refresh={jest.fn()} />
-      </Router>
-    )
-    const launchButton = await screen.findByText('Launch Audit')
-    fireEvent.click(launchButton, { bubbles: true })
-    await screen.findByText('Are you sure you want to launch the audit?')
-    const confirmLaunchButton = screen.getAllByText('Launch Audit')[1]
-    fireEvent.click(confirmLaunchButton, { bubbles: true })
-    await waitFor(() => {
-      expect(apiMock).toHaveBeenCalledTimes(5)
-      expect(apiMock.mock.calls[4][1]).toMatchObject({
-        body: JSON.stringify({
-          sampleSize: 46,
-          roundNum: 1,
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        method: 'POST',
-      })
+    auditSettingsMock.mockReturnValue(settingsMock.full)
+    const expectedCalls = [
+      apiCalls.getJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getTargetedContests,
+      apiCalls.getSampleSizeOptions,
+      apiCalls.postRound({ 'contest-id': 46 }),
+      apiCalls.getSampleSizeOptions,
```

Comment:
surprised that there's another call to `getSampleSizeOptions` after launching the round

Reply:
@jonahkagan That's due to me copy-pasting the pattern from one of the other api hooks, where you reload the data from the server after posting the update. In this context, though, I don't think that's necessary, so should change that probably.

Reply:
oh interesting, `withMockFetch` is supposed to fail if its given expected API calls that don't occur